jaxrs-clients/src/main/java/com/palantir/remoting/jaxrs/FeignJaxRsClientBuilder.java

```diff
@@ -107,12 +108,24 @@ private Contract createContract() {
     }
 
     private Encoder createEncoder(ObjectMapper objectMapper) {
-        // TODO(rfink) Get Jackson24Encoder if ObjectMapper#writerFor is unavailable
-        return new InputStreamDelegateEncoder(new TextDelegateEncoder(new JacksonEncoder(objectMapper)));
+        Encoder jacksonEncoder = hasJackson25()
+                ? new JacksonEncoder(objectMapper)
+                : new Jackson24Encoder(objectMapper);
+        return new InputStreamDelegateEncoder(new TextDelegateEncoder(jacksonEncoder));
+    }
+
+    // Uses reflection to determine if Jackson >= 2.5 is on the classpath by checking for the existence of the
+    // ObjectMapper#writerFor method.
+    private boolean hasJackson25() {
+        try {
+            ObjectMapper.class.getMethod("writerFor", Class.class);
```

Comment:
more specifically, it's going to try and use writerFor(JavaType)....but that's just a nit